lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -1241,22 +1241,48 @@ func (c *ContainerProxy) Remove(vc *viccontainer.VicContainer, config *types.Con
 // once the said container has been removed. It fetches a list of volumes that are joined
 // to at least one other container, and calls the portlayer to remove this container's
 // anonymous volumes if they are dangling. Errors, if any, are only logged.
-func removeAnonContainerVols(pl *client.PortLayer, cID string, volumes map[string]struct{}) {
+func removeAnonContainerVols(pl *client.PortLayer, cID string, vc *viccontainer.VicContainer) {
+	// NOTE: these strings come in the form of <volume id>:<destination>:<volume options>
+	volumes := vc.Config.Volumes
+	// NOTE: these strings come in the form of <volume id>:<destination path>
+	namedVolumes := vc.HostConfig.Binds
+
+	// assemble a mask of volume paths before processing binds. MUST be paths, as we want to move to honoring the proper metadata in the "volumes" section in the future.
+	namedMaskList := make(map[string]struct{}, 0)
+	for _, entry := range namedVolumes {
+		fields := strings.SplitN(entry, ":", 2)
+		if len(fields) >= 2 {
```

Comment:
Are we looking for `/path/to/volume` format here (vs. `name:/path/to/volume`)?  I'm curious as to why we error, but I confess I don't know the expected format of "proper metadata".

Reply:
Expected format is above in comments for the `volumes` and `namedVolumes` vars. This particular section should always be in the `<name>:<path>` format. this is the correct docker prescribed format for their metadata.

Reply:
@matthewavery then shouldn't it be `if len(fields) > 2` if we expect it to be 2? Currently you log an error on the `len(fields) == 2` case.